staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4163,6 +4182,17 @@ type ServiceSpec struct {
 	// +optional
 	ExternalTrafficPolicy ServiceExternalTrafficPolicyType `json:"externalTrafficPolicy,omitempty" protobuf:"bytes,11,opt,name=externalTrafficPolicy"`
 
+	// InternalTrafficPolicy specifies if the cluster internal traffic
+	// should be routed to cluster-wide endpoints, node-local endpoints only
+	// or both with node-local ones preferred if there is any. "Cluster"
+	// routes internal traffic to a Service to all cluster-wide endpoints.
+	// "PreferLocal" will route internal traffic to node-local endpoints if
+	// one exists, otherwise it will fallback to the same behavior as
+	// "Cluster". "Local" routes traffic to node-local endpoints only,
+	// traffic is dropped if no node-local endpoints are ready.
+	// +optional
+	InternalTrafficPolicy ServiceInternalTrafficPolicyType
```

Comment:
The published v1 types need json tags (see `externalTrafficPolicy` above for format). Need to use an unused protobuf number though (21 is the next one I believe)

Reply:
This also needs to be a string pointer